Example output:

```shell
 make kind-up 
./hack/kind-up.sh \
		--cluster-name gardener-local \
		--path-kubeconfig ./gardener/example/provider-local/seed-kind/base/kubeconfig \
		--path-cluster-values ./gardener/example/gardener-local/kind/local/values.yaml \
		--with-lpp-resize-support false \
		
Error: Required shell dependencies not met. Please refer to https://github.com/gardener/gardener/blob/master/docs/development/local_setup.md#macos-only-install-gnu-core-utilities:
    - Current sed version does not support the --version flag. Please ensure GNU sed is installed.
    - BSD tar detected. Please ensure GNU tar is installed.
    - BSD grep detected. Please ensure GNU grep is installed.
make: *** [kind-up] Error 1
```

--- a/hack/kind-up.sh
+++ b/hack/kind-up.sh
@@ -236,6 +236,10 @@ check_shell_dependencies() {
   fi
 
   if [[ "$OSTYPE" == "darwin"* ]]; then
+    if ! date --version >/dev/null 2>&1; then
+      errors+=("Coreutils not found. Please ensure coreutils are installed.")
+    fi
+
     if gzip --version 2>&1 | grep -q "Apple"; then
       errors+=("Apple built-in gzip utility detected. Please ensure GNU gzip is installed.")
     fi

--- a/hack/kind-up.sh
+++ b/hack/kind-up.sh
@@ -224,7 +224,7 @@ check_shell_dependencies() {
   errors=()
 
   if ! sed --version >/dev/null 2>&1; then
-    errors+=("Current sed version does not support the --version flag. Please ensure GNU sed is installed.")
+    errors+=("Current sed version does not support --version flag. Please ensure GNU sed is installed.")
   fi
 
   if tar --version 2>&1 | grep -q "bsdtar"; then
@@ -237,7 +237,7 @@ check_shell_dependencies() {
 
   if [[ "$OSTYPE" == "darwin"* ]]; then
     if ! date --version >/dev/null 2>&1; then
-      errors+=("Coreutils not found. Please ensure coreutils are installed.")
+      errors+=("Current date version does not support --version flag. Please ensure coreutils are installed.")
     fi
 
     if gzip --version 2>&1 | grep -q "Apple"; then

